library/spdm_requester_lib/libspdm_req_key_update.c

```diff
@@ -140,16 +140,17 @@ static libspdm_return_t libspdm_try_key_update(libspdm_context_t *spdm_context,
             spdm_context, &session_id, &spdm_response_size, (void **)&spdm_response);
 
         if (!LIBSPDM_STATUS_IS_ERROR(status)) {
-            if (spdm_response_size < sizeof(spdm_message_header_t)) {
-                status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
-            } else if (spdm_response->header.spdm_version != spdm_request->header.spdm_version) {
-                status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
-            } else if (spdm_response->header.request_response_code == SPDM_ERROR) {
+            if (spdm_response->header.request_response_code == SPDM_ERROR) {
```

Comment:
Here the check order is slightly changed to make sure ERROR response especially `RESPONSE_IF_READY` can be handled before the exact `spdm_key_update_response_t` response size check.